### PR TITLE
Solution parse_list_DOM

### PR DESCRIPTION
**Pull request by NazariiAlieksieiev**
[DEMO LINK](https://NazariiAlieksieiev.github.io/js_task_parse_list_DOM/)

### CODE REVIEW

File: src/scripts/main.js [hunk 1]
```diff
@@ -1,3 +1,44 @@
...
+  sortedEmployees.sort((emloyeeA, employeeB) =>
+    employeeB.dataset.salary - emloyeeA.dataset.salary);
+
+  for (let i = 0; i < sortedEmployees.length; i++) {
+    list.appendChild(sortedEmployees[i]);
+  }
+};
+
+function getEmployees(list) {
+  const Employees = [];
```
Comment (mykhalenych) [inline src/scripts/main.js:27]: bad naming
```suggestion
  const employees = [];
```

File: src/scripts/main.js [hunk 2]
```diff
@@ -1,3 +1,44 @@
 'use strict';
 
 // write code here
+
+const listOfEmployees = document.querySelector('ul');
+
+function sortList(list) {
+  const sortedEmployees = [];
+
+  for (let i = 0; i < list.children.length; i++) {
```
Comment (mykhalenych) [inline src/scripts/main.js:10]: use array method instead of for, sort will be better

File: src/scripts/main.js [hunk 3]
```diff
@@ -1,3 +1,44 @@
...
+      .replace('$', '')
+      .replace(',', '');
+
+    sortedEmployees.push(list.children[i]);
+  }
+
+  sortedEmployees.sort((emloyeeA, employeeB) =>
+    employeeB.dataset.salary - emloyeeA.dataset.salary);
+
+  for (let i = 0; i < sortedEmployees.length; i++) {
```
Comment (mykhalenych) [inline src/scripts/main.js:21]:
```suggestion
```

File: src/scripts/main.js [hunk 4]
```diff
@@ -1,3 +1,44 @@
...
+    list.appendChild(sortedEmployees[i]);
```
Comment (mykhalenych) [inline src/scripts/main.js:22]:
```suggestion
    list.append(...sortedEmployees);
```

File: src/scripts/main.js [hunk 5]
```diff
@@ -1,3 +1,44 @@
...
+  }
+};
+
+function getEmployees(list) {
+  const Employees = [];
+
+  for (let i = 0; i < list.children.length; i++) {
```
Comment (mykhalenych) [inline src/scripts/main.js:29]: use map instead

**Review by mykhalenych — CHANGES_REQUESTED** [review]

File: src/scripts/main.js [hunk 6]
```diff
@@ -1,3 +1,39 @@
...
+
+  [...list.children].map(employee => employees.push({
+    name: employee.textContent.trim(),
+    position: employee.dataset.position,
+    salary: employee.dataset.salary,
+    age: employee.dataset.age,
+  })
+  );
+
+  return employees;
```
Comment (IvanFesenko) [inline src/scripts/main.js:35]: map return new array, no need to create separate variable
```suggestion
  return [...list.children].map(employee => ({
    name: employee.textContent.trim(),
    position: employee.dataset.position,
    salary: employee.dataset.salary,
    age: employee.dataset.age,
  }));
```

File: src/scripts/main.js [hunk 7]
```diff
@@ -1,3 +1,39 @@
...
+      .replace('$', '')
+      .replace(',', '');
+
+    sortedEmployees.push(employee);
+  }
+
+  [...list.children].map(toNumber);
+
+  sortedEmployees.sort((emloyeeA, employeeB) =>
+    employeeB.dataset.salary - emloyeeA.dataset.salary);
```
Comment (IvanFesenko) [inline src/scripts/main.js:19]: Refactor without creating separate variable. You can see example below.

**Review by IvanFesenko — CHANGES_REQUESTED** [review]
Please refactor without separate variables

**Review by mykhalenych — APPROVED** [review]

**Pull request closed without merging**